examples/compute-boids/main.rs

```diff
@@ -0,0 +1,391 @@
+        &surface,
+        &wgpu::SwapChainDescriptor {
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+            format: wgpu::TextureFormat::Bgra8UnormSrgb,
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+            present_mode: wgpu::PresentMode::Vsync,
+        },
+    );
+
+
+    // create depth texture attachment
+
+    let depth_texture = device.create_texture(&wgpu::TextureDescriptor {
+        size: wgpu::Extent3d { 
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+            depth: 1,
+        },
+        mip_level_count: 1,
+        array_layer_count: 1,
+        sample_count: 1,
+        dimension: wgpu::TextureDimension::D2,
+        format: wgpu::TextureFormat::Depth24PlusStencil8,
+        usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+    });
+
+
+    // buffer for the three 2d triangle vertices of each instance
+
+    let vertex_buffer_data: Vec<f32> = [-0.01, -0.02, 0.01, -0.02, 0.00, 0.02].to_vec();
+    let vertices_buffer = device.create_buffer_with_data(vertex_buffer_data.as_bytes(), 
+        wgpu::BufferUsage::VERTEX | wgpu::BufferUsage::COPY_DST);
+
+
+    // buffer for simulation parameters uniform
+
+    let sim_param_data: Vec<f32> = [
+        0.04,  // deltaT
+        0.1,   // rule1Distance
+        0.025, // rule2Distance
+        0.025, // rule3Distance
+        0.02,  // rule1Scale
+        0.05,  // rule2Scale
+        0.005  // rule3Scale
+    ].to_vec();
+    let sim_param_buffer = device.create_buffer_with_data(sim_param_data.as_bytes(), 
+        wgpu::BufferUsage::UNIFORM | wgpu::BufferUsage::COPY_DST);
+
+
+    // buffer for all particles data of type [(posx,posy,velx,vely),...]
+
+    let mut initial_particle_data: Vec<f32> = vec![0.0; (4 * NUM_PARTICLES) as usize];
+    for i in 0..NUM_PARTICLES {
+        initial_particle_data[(4 * i + 0) as usize] = 2.0 * (rand::random::<f32>() - 0.5);
+        initial_particle_data[(4 * i + 1) as usize] = 2.0 * (rand::random::<f32>() - 0.5);
+        initial_particle_data[(4 * i + 2) as usize] = 2.0 * (rand::random::<f32>() - 0.5) * 0.1;
+        initial_particle_data[(4 * i + 3) as usize] = 2.0 * (rand::random::<f32>() - 0.5) * 0.1;
+    }
+
+
+    // creates two buffers of particle data each of size NUM_PARTICLES
+    // the two buffers alternate as dst and src for each frame
+
+    let mut particle_buffers = Vec::<wgpu::Buffer>::new();
+    for _i in 0..2 {
+        particle_buffers.push(
+            device.create_buffer_with_data(initial_particle_data.as_bytes(), wgpu::BufferUsage::VERTEX
+                | wgpu::BufferUsage::STORAGE
+                | wgpu::BufferUsage::COPY_DST)
+        );
+    }
+    
+    
+    // create compute bind group for each particle buffer
+
+    let mut particle_bind_groups = Vec::<wgpu::BindGroup>::new();
+    for i in 0..particle_buffers.len() {
+        particle_bind_groups.push(
```

Comment:
can we do this in the same loop where we are creating the buffers?

Reply:
@kvark In this case we can't because each binding group references both buffers, so they must both be created ahead of time.